### PR TITLE
add task nextstrain_ncov_sanitize_gisaid_data and workflow sarscov2_gisaid_ingest

### PR DESCRIPTION
**Pull request by tomkinsc**
add task `nextstrain_ncov_sanitize_gisaid_data` and workflow `sarscov2_gisaid_ingest` to sanitize data downloaded directly from GISAID and transform to input suitable for the Nextstrain/augur pipeline. 

This carries out the processing (but not download) steps detailed here:
https://nextstrain.github.io/ncov/data-prep#curate-data-from-the-full-gisaid-database

### CODE REVIEW

**Review by dpark01 — COMMENTED** [review]
Looks conceptually good to me. Have we tried it yet?

Also, I think I'd prefer if the task could accept the `.tar.xz` files directly that gisaid tends to give.

**Comment by tomkinsc** [comment]
Thanks for catching that; fixed.

**Comment by tomkinsc** [comment]
I've tried it [here](https://job-manager.dsde-prod.broadinstitute.org/jobs/9b1dcb24-13f4-471f-9f0c-7dc8a8cc1473). Takes a while but seems to work.

**Comment by tomkinsc** [comment]
As we currently have our [builds](https://github.com/broadinstitute/nextstrain-builds/tree/0d46f0f821337a623209a63fc33a499ed0c2b634/configs) configured, the `sarscov2_nextstrain` workflow fails since the metadata lacks a column for `originating_lab`. GISAID does not currently include that info in the main metadata download, but hopefully that will change.

File: pipes/WDL/tasks/tasks_nextstrain.wdl [hunk 1]
```diff
@@ -376,6 +376,68 @@ task nextstrain_ncov_defaults {
...
+        --parse-location-field Location \
+        --rename-fields 'Virus name=strain' 'Accession ID=gisaid_epi_isl' 'Collection date=date' 'Clade=GISAID_clade' 'Pango lineage=pango_lineage' 'Host=host' 'Type=virus' 'Patient age=age' \
+        ~{"--strip-prefixes=" + prefix_to_strip} \
+        --output "~{out_basename}_metadata_sanitized_for_nextstrain.tsv.gz"
+    }
+    runtime {
+        docker: docker
+        memory: "7 GB"
+        cpu:   1
+        disks:  "local-disk 50 HDD"
```
Comment (dpark01) [inline pipes/WDL/tasks/tasks_nextstrain.wdl:432]: I'm a bit surprised we can get away with 50GB disk, but given that cloud disk speeds are proportional to disk size, it seems we should ask for more anyway. Or maybe just get around the whole thing with a `disks: "local-disk 375 LOCAL"` to max out the I/O on a GCE VM.

File: pipes/WDL/workflows/sarscov2_gisaid_ingest.wdl [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            description: "Tab-separated metadata file for sequences downloaded from GISAID and passed in via sequences_gisaid_fasta.",
+            patterns: ["*.txt", "*.tsv","*.tsv.xy","*.tsv.gz"]
+        }
+    }
+    call nextstrain.nextstrain_ncov_sanitize_gisaid_data as sanitize_gisaid {
+        input:
+            sequences_gisaid_fasta = sequences_gisaid_fasta,
+            metadata_gisaid_tsv    = metadata_gisaid_tsv
+    }
+    output {
```
Comment (dpark01) [inline pipes/WDL/workflows/sarscov2_gisaid_ingest.wdl:28]: hey let's add in a `call` to `terra.gcs_copy` if an optional `String?` gcs bucket path is specified… I can imagine a typical use pattern where we download stuff from gisaid, run this, and push it back to our private bucket for future internal use

**Review by dpark01 — APPROVED** [review]
> I've tried it [here](https://job-manager.dsde-prod.broadinstitute.org/jobs/9b1dcb24-13f4-471f-9f0c-7dc8a8cc1473). Takes a while but seems to work.

Cool… so you're directly consuming the (unfortunately structured) tsv.tar.xz and fasta.tar.xz files. I think `xz` is known to be slow, so that's not terribly surprising. I suggested a change to your runtime block that might help on the I/O side a bit, but it's still not going to be as fast as zstd.

**Pull request merged**